### ISSUE TITLE
Project Proposal

### ISSUE THREAD

**Issue by Nackles** [QuietEvolver/RecipEasy#10]
PROPOSAL: RecipEasy
 
Project Title: RecipEasy
Team Members: Vera, Micah, Nick, Claire
Project Description: App allows user to search for and select a recipe, quickly navigate and track the ingredients needed, and find the nearest grocery store bypassing long narrative and a separate look up of where to shop based on user preferences.
 
APIs:
    Recipe API: Spoontacular
    Store Locator API: Google Maps URLs
Possibly implement Google Maps Static
Possibly implement Google Maps Imbedded API
 
Sketch/Mockup:
 
 
Rough Breakdown of Tasks:  
Front-end: Claire, Micah
Back-end: Nick, Vera
Master master: Vera
Project Progress Tracking: Claire and Micah (slack, trello, git hub, which combo is best?)
  
 
MVP
 
Idea: Three step app:
       	1-Select a recipe
       	2-Make an ingredient list
       	3-Find a grocery store location
       	4-See recipe instructions to make meal
*these are the main components that the app will be based upon. Actual flow and transitions may be less rigid.
Target Audience: Anyone who cooks, travelers, people who like efficiency
Problem: When pulling up a recipe online these days, you have to read a bloggers whole life story just to get down to what they are actually making and what ingredients you need.  Skip the novel and get straight to the goods with RecipEasy; locate a delish recipe, track exactly what ingredients you need, and see where to go to get the goods so that you can get cooking in record time.
Goal: Find recipe. List needed ingredients. Find location to purchase ingredients. Store ingredients list as *haves* vs *needs to buy*. See how to make dish.
 
User Stories:
·       User loads RecipEasy application and sees beautiful landing page
·       User selects a recipe for what they want to make
o  Genre selector cards or list
o  Custom search field
·       App displays ingredient list (taken from selected recipe) and presents it in the form of a grocery list with selectable elements
·       User checks off what items they may already have on hand
·       App reorders list to show unchecked boxes at the top of the list once user is done selecting
o   Potential button to hit for I’m done ‘let’s get shopping’
·       User is presented with a list of local grocery stores to pick up remaining items on grocery list
o   List is populated based on the user's current location
·       Once all items on the list are checked (or a button is selected ‘let’s get cooking’), the cooking instructions display in the app
 
User Stories Secondary:
What does the user do?
 
User loads app
User types a recipe and hits search
Options may include: avoid allergens, use dairy alternative, etc.
App populates with recipes w/ ~information.
User selects a recipe
Other recipes hide.
Selected recipe displays in full.
User hits go
Ingredients transfer to interactive *Needs* vs *Haves*.
User selects an ingredient
Ingredient toggles between *Need* and *Have*.
User selects Map option
Google Maps URL API opens their OS’s Google Maps App/browser with directions to “nearest grocery store”.
User can then continue with directions in Maps and return to RecipEasy when ready to shop.
 
Nice-To-Haves/What might user want?
·       ***ALL STAR*** App displays grocery locations in the order of which location has which items (or the most items) on the remaining grocery list
·      Option to override location in case you are looking from a location other than where you wish to shop (aka looking at the airport before flying to your destination)
·       Ability to input any allergies before searching for recipes to weed out recipes that contain those ingredients
·       Ability to save recipes and record number of times cooked and feedback.
·       App loads last results/search state from previous session.
·       Ability to go back and star or ‘unfollow’ recipes depending on how much you enjoyed them to keep a running list and store on a database.
·       Wine/beverage pairing suggestions that compliment the user's recipe selection
·      Ability to show video of how to make meal.
 
Potential components used:
       	·   HTML
       	·   JavaScript
       	·   CSS
       	·   Bootstrap
·   Recipe API: spoontacular
       	·   Grocery Location: Google Maps API
       	·   Algolia Places (JS library for addresses)
       	·   Multiple JS (JS library for visual effects)
